### PR TITLE
Use priority mutex for mutable state cache lock

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Deprecate locks.Mutex in favor of locks.PriorityMutex
* Wire history service logic using locks.PriorityMutex
  * API side will use high priority lock
  * Task processing side will use lower priority lock
* Remove some unnecessary complex logic

**Why?**
`locks.PriorityMutex` provides better performance then `locks.Mutex`, see #1611 

**How did you test it?**
Existing tests & CICD

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/nDCActivityReplicator.go [hunk 1]
```diff
@@ -98,7 +98,12 @@ func (r *nDCActivityReplicatorImpl) SyncActivity(
 		RunId:      request.RunId,
 	}
 
-	context, release, err := r.historyCache.getOrCreateWorkflowExecution(ctx, namespaceID, execution)
+	context, release, err := r.historyCache.getOrCreateWorkflowExecution(
+		ctx,
+		namespaceID,
+		execution,
+		callerTypeAPI,
```
Comment (yiminc) [inline service/history/nDCActivityReplicator.go:105]: should this be callerTypeTask?

**Review by yiminc — COMMENTED** [review]
Should the replication be treated as low priority compare to API calls?

File: service/history/nDCActivityReplicator.go (hunk 1 above)
Comment (wxing1292) [reply service/history/nDCActivityReplicator.go:105]: this is for NDC activity replication, meaning usually this API is invoked by the passive side.
for passive side, replication API calls should be placed above back ground task processing.

Comment (yiminc) [reply service/history/nDCActivityReplicator.go:105]: Is replicatorQueueProcessor.go also only happen on passive side?

Comment (wxing1292) [reply service/history/nDCActivityReplicator.go:105]: maybe some background: target (passive) cluster will call source cluster for replication tasks, this is done by periodically polling source cluster

**Review by yiminc — APPROVED** [review]

**Pull request merged**